File: pkg/api/dtos/catalog.go
package dtos

// {
//     "name": "string",
//     "components": [{
//         "name": "string",
//         "labels": [{"key": "value"}],
//         "team": "string",
//         "namespace": "string",
//         "address": "string",
//         "pods": [{
//             "name": "string",
//             "status": "green|yellow|red"
//         }],
//         "external": "boolean"
//     }]
// }

type Catalog []CatalogService

type CatalogService struct {
	Name       string             `json:"name,omitempty"`
	Components []CatalogComponent `json:"components,omitempty"`
}

type CatalogLabels map[string]string

type CatalogPodStatus int

const (
	PodStatusGreen CatalogPodStatus = iota
	PodStatusYellow
	PodStatusRed
)

type CatalogPod struct {
	Name   string           `json:"name,omitempty"`
	Status CatalogPodStatus `json:"status,omitempty"`
}

type CatalogComponent struct {
	Name      string        `json:"name,omitempty"`
	Labels    CatalogLabels `json:"labels,omitempty"`
	Teams     []string      `json:"teams,omitempty"`
	Namespace string        `json:"namespace,omitempty"`
	Address   string        `json:"address,omitempty"`
	Pods      []CatalogPod  `json:"pods,omitempty"`
	External  bool          `json:"external"`
}

File: public/app/types/catalog.ts
export type Catalog = CatalogService[];

export interface CatalogService {
  name: string;
  components: ServiceComponent[];
}

export type CatalogLabels = {
  [key: string]: string;
};

export enum CatalogPodStatus {
  Green = 0,
  Yellow = 1,
  Red = 2,
}

export interface CatalogPod {
  name: string;
  status: CatalogPodStatus;
}

export interface ServiceComponent {
  name: string;
  labels?: CatalogLabels;
  teams: string[];
  namespace: string;
  address: string;
  pods?: CatalogPod[];
  external: boolean;
}

export interface CatalogState {
  catalog: Catalog;
}
